tasks/libs/common/remote_api.py

```diff
@@ -59,24 +62,30 @@ def request(
         try:
             # If json_input is true, we specifically want to send data using the json
             # parameter of requests.post / requests.put
-            if method == "PUT":
-                if json_input:
-                    r = requests.put(url, headers=headers, json=data, stream=stream_output)
+            for retry_count in range(self.requests_500_retry_count):
+                if method == "PUT":
+                    if json_input:
+                        r = requests.put(url, headers=headers, json=data, stream=stream_output)
+                    else:
+                        r = requests.put(url, headers=headers, data=data, stream=stream_output)
+                elif method == "DELETE":
+                    r = requests.delete(url, headers=headers, stream=stream_output)
+                elif data or method == "POST":
+                    if json_input:
+                        r = requests.post(url, headers=headers, json=data, stream=stream_output)
+                    else:
+                        r = requests.post(url, headers=headers, data=data, stream=stream_output)
                 else:
-                    r = requests.put(url, headers=headers, data=data, stream=stream_output)
-            elif method == "DELETE":
-                r = requests.delete(url, headers=headers, stream=stream_output)
-            elif data or method == "POST":
-                if json_input:
-                    r = requests.post(url, headers=headers, json=data, stream=stream_output)
+                    r = requests.get(url, headers=headers, stream=stream_output)
+                if r.status_code >= 400:
+                    if r.status_code == 401:
+                        print(self.authorization_error_message)
+                    elif 500 <= r.status_code < 600:
+                        time.sleep(self.requests_sleep_time + retry_count * self.requests_sleep_time)
```

Comment:
Nit: for clarity, we should print a message before the `time.sleep` when we retry, like `Request failed with error <status code>, retrying in <duration> seconds (retry <retry n°>/<total retry n°>)`.